tests/SpecTests/Operation.php

```diff
@@ -500,6 +504,35 @@ private function executeForSession(Session $session, Context $context)
         }
     }
 
+    private function executeForTestRunner(FunctionalTestCase $test, Context $context)
+    {
+        switch ($this->name) {
+            case 'assertSessionPinned':
+                $session = $context->{$this->arguments['session']};
+                $test->assertNotNull($session->getServer());
+
+                return null;
+            case 'assertSessionTransactionState':
+                $session = $context->{$this->arguments['session']};
+                $expected = in_array($this->arguments['state'], ['in_progress', 'starting']);
+                $test->assertSame($expected, $session->isInTransaction());
```

Comment:
I previously brought up this issue for PHP in https://github.com/mongodb/specifications/pull/548#discussion_r285253695 for [SPEC-1195](https://jira.mongodb.org/browse/SPEC-1195). I think this is a good compromise, but we'd probably do well to leave a TODO and create a new PHPC ticket to suggest implementing some API to expose the session state. Whether we ultimately decide to do that, and lobby for the necessary libmongoc change, can be decided later but it'd be good to record the proposal and any decision in a formal record.

Reply:
I've added a comment and created [PHPC-1438](https://jira.mongodb.org/browse/PHPC-1438) to track exposing the internal transaction state.